spec/truffle/arguments_descriptor_spec.rb

```diff
@@ -102,45 +102,73 @@ def posn_kws_defaults(a, b: 101, c: 102)
   it "contain optional keyword arguments with positional arguments" do
     info = posn_kws_defaults(1, b: 2, c: 3)
     info.values.should == [1, 2, 3]
-    info.descriptor.should == [:keywords, :b, :c] if truffleruby?
+    info.descriptor.should == [:b, :c, :no_splat] if truffleruby?
     info.arguments.should == [1, {b: 2, c: 3}] if truffleruby?
   end
 
   it "do not contain missing optional keyword arguments" do
     info = posn_kws_defaults(1, b: 2)
     info.values.should == [1, 2, 102]
-    info.descriptor.should == [:keywords, :b] if truffleruby?
+    info.descriptor.should == [:b, :no_splat] if truffleruby?
     info.arguments.should == [1, {b: 2}] if truffleruby?
 
     info = posn_kws_defaults(1, c: 3)
     info.values.should == [1, 101, 3]
-    info.descriptor.should == [:keywords, :c] if truffleruby?
+    info.descriptor.should == [:c, :no_splat] if truffleruby?
     info.arguments.should == [1, {c: 3}] if truffleruby?
   end
 
   it "do not contain the name of distant explicitly splatted keyword arguments" do
     distant = {b: 2}
     info = only_kws(a: 1, **distant)
     info.values.should == [1, 2]
-    info.descriptor.should == [:keywords, :a] if truffleruby?
+    info.descriptor.should == [:a, :post_splat] if truffleruby?
     info.arguments.should == [{a: 1, b: 2}] if truffleruby?
   end
 
   it "do not contain the name of near explicitly splatted keyword arguments" do
     info = only_kws(a: 1, **{b: 2})
     info.values.should == [1, 2]
-    info.descriptor.should == [:keywords, :a] if truffleruby?
+    info.descriptor.should == [:a, :post_splat] if truffleruby?
     info.arguments.should == [{a: 1, b: 2}] if truffleruby?
   end
 
+  it "overwrite earlier splatted keyword arguments" do
+    splatted = {b: 3}
+    info = only_kws(**splatted, a: 1, b: 2)
+    info.values.should == [1, 2]
+    info.descriptor.should == [:a, :b, :pre_splat] if truffleruby?
+    info.arguments.should == [{a: 1, b: 2}] if truffleruby?
+  end
+
+  it "do not overwrite later splatted keyword arguments" do
+    splatted = {b: 3}
+    info = only_kws(a: 1, b: 2, **splatted)
+    info.values.should == [1, 3]
+    info.descriptor.should == [:a, :b, :post_splat] if truffleruby?
+    info.arguments.should == [{a: 1, b: 3}] if truffleruby?
+  end
+
+  it "cause correct copy of an only-splat" do
+    def kwargs(**kwargs)
+      info(kwargs, Primitive.arguments_descriptor, Primitive.arguments)
+    end
+
+    splatted = {b: 3}
+    info = kwargs(**splatted)
+    info.values.should_not.equal?(splatted)
+    info.descriptor.should == [:only_splat] if truffleruby?
+    info.arguments[0].should.equal?(splatted) if truffleruby?
+  end
```

Comment:
Could you add a spec for a splat in the middle? `kwargs(a: 1, **kw, c: 3)`

Reply:
I think one way to handle this is to either consider all of it is a splat ONLY_SPLAT and in the translator already introduce the nodes to merge and have a single rest kw arg, or e.g. consider it a PRE_SPLAT and in the translator merge the trailer in the kw splat.